src/nlp/api.jl

```diff
@@ -131,6 +135,7 @@ Evaluate ``f(x)`` and ``c(x)`` at `x`. `c` is overwritten with the value of ``c(
 function objcons!(nlp::AbstractNLPModel, x::AbstractVector, c::AbstractVector)
   @lencheck nlp.meta.nvar x
   @lencheck nlp.meta.ncon c
+  check_unconstrained(nlp)
```

Comment:
Is this one necessary? The call to `cons()` will also check.

Reply:
I added it everywhere in the same spirit of the lencheck.
#467 would call objcons! by the way